src/BenchmarksClient/Startup.cs

```diff
@@ -100,72 +100,91 @@ public static int Main(string[] args)
+                                await worker.StartJobAsync(job);
                             }
                         }
                         catch (Exception e)
                         {
                             Log($"An unexpected error occured while starting the job {job.Id}");
                             Log(e.ToString());
 
-                            job.State = ClientState.Deleting;
+                            job.State = ClientJobState.Deleting;
                         }
                     }
-                    else if (job.State == ClientState.Running || job.State == ClientState.Completed)
+                    else if (job.State == ClientJobState.Running || job.State == ClientJobState.Completed)
                     {
                         var now = DateTime.UtcNow;
 
                         // Clean the job in case the driver is not running
                         if (now - job.LastDriverCommunicationUtc > TimeSpan.FromSeconds(30))
                         {
                             Log($"Driver didn't communicate for {now - job.LastDriverCommunicationUtc}. Halting job.");
-                            job.State = ClientState.Deleting;
+                            job.State = ClientJobState.Deleting;
                         }
                     }
-                    else if (job.State == ClientState.Deleting)
+                    else if (job.State == ClientJobState.Deleting)
                     {
                         Log($"Deleting job {worker?.JobLogText ?? "no worker found"}");
 
                         try
                         {
                             if (worker != null)
                             {
-                                await worker.StopAsync();
+                                // We stop the worker but 
+                                await worker.StopJobAsync();
                             }
                         }
                         finally
                         {
-                            worker?.Dispose();
-                            worker = null;
-
                             _jobs.Remove(job.Id);
                         }
                     }
                 }
                 await Task.Delay(100);
+
+                allJobs = _jobs.GetAll();
+                job = allJobs.FirstOrDefault(clientJob =>
+                {
+                    return string.Equals(clientJob.SpanId, job.SpanId);
+                });
+                // If no more jobs with the span id exist then we can clear
+                // out the worker to signal to the worker factory to create
+                // a new worker.
+                if (job == null && worker != null)
+                {
+                    await worker.DisposeAsync();
+                    worker = null;
+
+                    // Get another job for the new worker we are going to create
+                    job = allJobs.FirstOrDefault();
```

Comment:
If there are no other jobs in the queue yet you'll never get a new job it looks like